tock/tock/templates/base.html

```diff
@@ -37,6 +37,35 @@
     ga('set', 'forceSSL', true);
     ga('send', 'pageview');
   </script>
+  <script>
+      function extendSession() {
+        var request = new XMLHttpRequest();
+        request.open("GET", "/session-extend", true);
+        request.onload = function () {
+          if (request.status === 401) {
+            alert("Your session has expired. Please log back in.");
+            logout();
+          }
+        };
+        request.send();
+        setTimeout(setAlert, timeoutCount - 120 * 1000);
+      }
+
+      function logout() {
+        window.location.reload();
+      }
+
+      const timeoutCount = "{{x_tock_session_timeout_ms}}";
+
+      function setAlert() {
+        alert(
+          "Your session is about to expire! We will attempt to refresh your session."
+        );
+        extendSession();
+      }
+
+      setTimeout(setAlert, timeoutCount - 120 * 1000);
```

Comment:
Do we need to modify `timeoutCount` here? I think we're already rendering it on the template in milliseconds.

Reply:
Ah I see, we're firing the alert 2 minutes prior to the timeout.